### ISSUE TITLE
Statping dashboard missing header

### ISSUE THREAD

**Issue by florian-besser** [statping-ng/statping-ng#283]
Locally I can run:
```
docker run -it -p 8080:8080 ghcr.io/statping-ng/statping-ng:0.91.0
```

This allows me to access /dashboard, which looks correct:
![image](https://github.com/statping-ng/statping-ng/assets/7090706/23337c73-4124-4322-863d-3e8b222b4146)

I wanted to use `statping-ng` in our test environment to see how it would perform; as such I ran a `k8s` deployment configured like so:
```
  statping:
    Container ID:   containerd://e8199351b6836a8d5537c384d28a0bf519b3643ba715d18ebd39308fd55bf05c
    Image:          ghcr.io/statping-ng/statping-ng:0.91.0
    Image ID:       ghcr.io/statping-ng/statping-ng@sha256:50ba35d79ebac36ead6bb1473f59ee5718b16dc80328d8052edb139f57f823e9
    Port:           8080/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Thu, 28 Mar 2024 15:53:30 +0800
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     50m
      memory:  128Mi
    Requests:
      cpu:     50m
      memory:  128Mi
    Environment:
      PORT:              8080
      DB_CONN:           postgres
      DB_HOST:           (censored)
      DB_PORT:           5432
      DB_USER:           statping
      DB_PASS:           (censored)
      DB_DATABASE:       statping
      POSTGRES_SSLMODE:  require
      READ_ONLY:         false
      DISABLE_COLORS:    true
      REMOVE_AFTER:      8760h
      CLEANUP_INTERVAL:  24h
      ADMIN_USER:        admin
      ADMIN_PASSWORD:    (censored)
      SAMPLE_DATA:       false
      NAME:              (censored)
      DESCRIPTION:       (censored)
      DOMAIN:            (censored)
      API_SECRET:        (censored)
    Mounts:
      /app/services.yml from services (rw,path="services.yml")
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-2n9qj (ro)

```

When I move to `/dashboard` on that instance the screenshot shows that the header is missing
![image](https://github.com/statping-ng/statping-ng/assets/7090706/6f177a4c-e4e0-4dc1-8867-795f896589a2)

This means I cannot navigate to other parts of the dashboard, e.g. settings.

If I manually navigate to `dashboard/settings` in both cases the local installation again looks correct:
![image](https://github.com/statping-ng/statping-ng/assets/7090706/465ebb10-6284-4a7e-8102-50faaee8aa90)

Whereas the test instance again is missing the header:
![image](https://github.com/statping-ng/statping-ng/assets/7090706/d098345a-7f01-4aa5-8c95-b98b65cd7777)

The logs are not helpful to me, as no issue is highlighted.
```
time="2024-03-28T07:53:30Z" level=info msg="Starting Statping v0.91.0" type=cmd
time="2024-03-28T07:53:30Z" level=info msg="Attempting to read config file at: /app/config.yml" type=configs
time="2024-03-28T07:53:31Z" level=info msg="Database postgres connection was successful." type=configs
time="2024-03-28T07:53:31Z" level=info msg="Migrating Database Tables..." type=configs
time="2024-03-28T07:53:32Z" level=info msg="Migrating App to version: v0.91.0 (66a1adaa66acfa9615d5b8661e3c0320109731c3)" type=configs
time="2024-03-28T07:53:32Z" level=info msg="Statping Database Tables Migrated" type=configs
time="2024-03-28T07:53:32Z" level=info msg="Database Indexes Created" type=configs
time="2024-03-28T07:53:32Z" level=info msg="Error Reporting initiated, thank you!"
time="2024-03-28T07:53:34Z" level=info msg="Starting monitoring process for 13 Services" type=service
time="2024-03-28T07:53:35Z" level=info msg="Found 13 services inside services.yml file" type=service
time="2024-03-28T07:53:35Z" level=info msg="Statping HTTP Server running on http://0.0.0.0:8080/" type=handlers
time="2024-03-28T07:53:35Z" level=info msg="Database Cleanup runs every 24h0m0s and will remove records older than 8760h0m0s" type=database
```

Any thoughts on how to fix or debug this issue?

**Comment by jemand771** [comment]
oh that's funny 😄

can you take a look at your browser's network devtools? I assume some requests aren't going through to statping or erroring out on the way
also - fellow kubernetes enjoyer here - if you send your full manifest (so deployment+service+ingress, if that's your setup) I can try to reproduce this

**Comment by florian-besser** [comment]
Thanks for the quick response!

The network tab only contains `200 OK` answers:
![image](https://github.com/statping-ng/statping-ng/assets/7090706/7816c85e-678a-40aa-a228-8943f1d1fcba)
![image](https://github.com/statping-ng/statping-ng/assets/7090706/f22c4f39-4eb2-4f11-b50e-0c68016db1dd)

We use Terraform for our infrastructure as code stuff, let me paste you what we use:
```
resource "kubernetes_config_map" "statping_services" {
  metadata {
    name      = "statping-services"
    namespace = "alm"
  }

  data = {
    "services.yml" = <<-EOT
x-httpservice: &httpservice
  type: http
  method: GET
  check_interval: 60
  timeout: 10
  expected_status: 200
  allow_notifications: true
  notify_after: 2
  notify_all_changes: true
  public: true
  redirect: true
services:
  - name: Network Mesh
    domain: http://linkerd-identity.linkerd.svc.cluster.local:9990/metrics
    <<: *httpservice
  - name: Alerting
    domain: http://prometheus-server.alm.svc.cluster.local:80/-/ready
    <<: *httpservice
  - name: Kubernetes
    domain: https://kubernetes.default.svc.cluster.local:443/readyz
    <<: *httpservice
EOT
  }
}

resource "kubernetes_deployment" "statping" {
  metadata {
    name      = "statping"
    namespace = "alm"
    labels = {
      "app" : "statping"
    }
  }
  spec {
    replicas = 1
    selector {
      match_labels = {
        "app" : "statping"
      }
    }
    template {
      metadata {
        labels = {
          "app" : "statping"
        }
      }
      spec {
        container {
          name              = "statping"
          image             = "ghcr.io/statping-ng/statping-ng:0.91.0"
          image_pull_policy = "IfNotPresent"
          #          command = ["/bin/sh", "-c", "statping --port $PORT"]
          resources {
            limits = {
              cpu : "50m"
              memory : "128Mi"
            }
            requests = {
              cpu : "50m"
              memory : "128Mi"
            }
          }
          port {
            name           = "http"
            container_port = 8080
            protocol       = "TCP"
          }
          env {
            name  = "PORT"
            value = "8080"
          }

          env {
            name  = "DB_CONN"
            value = "postgres"
          }
          env {
            name  = "DB_HOST"
            value = (censored)
          }
          env {
            name  = "DB_PORT"
            value = "5432"
          }
          env {
            name  = "DB_USER"
            value = "statping"
          }
          env {
            name  = "DB_PASS"
            value = (censored)
          }
          env {
            name  = "DB_DATABASE"
            value = "statping"
          }
          env {
            name  = "POSTGRES_SSLMODE"
            value = "require"
          }
          env {
            name  = "READ_ONLY"
            value = "false"
          }
          env {
            name  = "DISABLE_COLORS"
            value = "true"
          }
#          env {
#            name  = "VERBOSE"
#            value = "4"
#          }
          env {
            name  = "REMOVE_AFTER"
            value = "8760h" # 1 year
          }
          env {
            name  = "CLEANUP_INTERVAL"
            value = "24h"
          }
          env {
            name  = "ADMIN_USER"
            value = "admin"
          }
          env {
            name  = "ADMIN_PASSWORD"
            value = random_password.statping_admin_pw.result
          }
          env {
            name  = "SAMPLE_DATA"
            value = "false"
          }
          env {
            name  = "NAME"
            value = (censored)
          }
          env {
            name  = "DESCRIPTION"
            value = (censored)
          }
          env {
            name  = "DOMAIN"
            value = (censored)
          }
          env {
            name  = "API_SECRET"
            value = random_password.statping_api_secret.result
          }
          volume_mount {
            name       = "services"
            mount_path = "/app/services.yml"
            sub_path   = "services.yml"
          }
        }
        volume {
          name = "services"
          config_map {
            name = kubernetes_config_map.statping_services.metadata[0].name
          }
        }
      }

    }
  }
  timeouts {
    create = "2m"
    update = "2m"
  }
}

resource "random_password" "statping_admin_pw" {
  length  = 32
  special = false
}

resource "random_password" "statping_api_secret" {
  length  = 128
  special = false
}

resource "kubernetes_service" "statping" {
  metadata {
    name      = "statping"
    namespace = "alm"
    labels = {
      "app" : "statping"
    }
  }
  spec {
    type = "ClusterIP"
    port {
      port        = 8080
      target_port = 8080
      protocol    = "TCP"
    }
    selector = {
      "app" : "statping"
    }
  }
}

resource "kubernetes_ingress_v1" "statping" {
  metadata {
    name      = "public"
    namespace = "alm"
    annotations = {
      # Allow world-wide ingress
      "nginx.ingress.kubernetes.io/whitelist-source-range" : "0.0.0.0/0"
    }
  }

  spec {
    ingress_class_name = "nginx"
    rule {
      host = (censored)
      http {
        path {
          backend {
            service {
              name = "statping"
              port {
                number = 8080
              }
            }
          }
        }
      }
    }
  }
}
```

Please note a few things on the above:
- `kubernetes_ingress_v1` requires some additional setup in your K8s cluster; consider using the `service` directly for quicker debugging.
- `kubernetes_deployment` relies on a Postgres database to be already available and configured correctly (e.g. having a `statping` database and a `statping` user)

Since I don't know if you like Terraform or not let me share the same info in K8s format as well.
Deployment:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    deployment.kubernetes.io/revision: "19"
  creationTimestamp: "2024-01-19T07:21:36Z"
  generation: 21
  labels:
    app: statping
  name: statping
  namespace: alm
  resourceVersion: "235807594"
  uid: 7971fa1b-e9a8-449d-ad15-eb4cca9af389
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: statping
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: statping
    spec:
      automountServiceAccountToken: true
      containers:
      - env:
        - name: PORT
          value: "8080"
        - name: DB_CONN
          value: postgres
        - name: DB_HOST
          value: (censored)
        - name: DB_PORT
          value: "5432"
        - name: DB_USER
          value: statping
        - name: DB_PASS
          value: (censored)
        - name: DB_DATABASE
          value: statping
        - name: POSTGRES_SSLMODE
          value: require
        - name: READ_ONLY
          value: "false"
        - name: DISABLE_COLORS
          value: "true"
        - name: REMOVE_AFTER
          value: 8760h
        - name: CLEANUP_INTERVAL
          value: 24h
        - name: ADMIN_USER
          value: admin
        - name: ADMIN_PASSWORD
          value: (censored)
        - name: SAMPLE_DATA
          value: "false"
        - name: NAME
          value: (censored)
        - name: DESCRIPTION
          value: (censored)
        - name: DOMAIN
          value: (censored)
        - name: API_SECRET
          value: (censored)
        image: ghcr.io/statping-ng/statping-ng:0.91.0
        imagePullPolicy: IfNotPresent
        name: statping
        ports:
        - containerPort: 8080
          name: http
          protocol: TCP
        resources:
          limits:
            cpu: 50m
            memory: 128Mi
          requests:
            cpu: 50m
            memory: 128Mi
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /app/services.yml
          mountPropagation: None
          name: services
          subPath: services.yml
      dnsPolicy: ClusterFirst
      enableServiceLinks: true
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      shareProcessNamespace: false
      terminationGracePeriodSeconds: 30
      volumes:
      - configMap:
          defaultMode: 420
          name: statping-services
          optional: false
        name: services
status:
  availableReplicas: 1
  conditions:
  - lastTransitionTime: "2024-03-28T07:35:40Z"
    lastUpdateTime: "2024-03-28T07:35:40Z"
    message: Deployment has minimum availability.
    reason: MinimumReplicasAvailable
    status: "True"
    type: Available
  - lastTransitionTime: "2024-01-19T07:21:36Z"
    lastUpdateTime: "2024-03-28T07:53:31Z"
    message: ReplicaSet "statping-7bb7c8985c" has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  observedGeneration: 21
  readyReplicas: 1
  replicas: 1
  updatedReplicas: 1
```

Service:
```
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: "2024-01-19T06:58:30Z"
  labels:
    app: statping
  name: statping
  namespace: alm
  resourceVersion: "202872668"
  uid: eb2c014f-4f21-4220-b33e-d052d1142292
spec:
  clusterIP: 172.20.179.77
  clusterIPs:
  - 172.20.179.77
  internalTrafficPolicy: Cluster
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - port: 8080
    protocol: TCP
    targetPort: 8080
  selector:
    app: statping
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```

Ingress:
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    nginx.ingress.kubernetes.io/whitelist-source-range: 0.0.0.0/0
  creationTimestamp: "2024-01-19T04:57:33Z"
  generation: 1
  name: public
  namespace: alm
  resourceVersion: "202826242"
  uid: 51ef0418-b202-45b6-ae25-5cee85de20f3
spec:
  ingressClassName: nginx
  rules:
  - host: (censored)
    http:
      paths:
      - backend:
          service:
            name: statping
            port:
              number: 8080
        pathType: ImplementationSpecific
status:
  loadBalancer:
    ingress:
    - hostname: (censored)
```

Config-map:
```
apiVersion: v1
data:
  services.yml: |
    x-httpservice: &httpservice
      type: http
      method: GET
      check_interval: 60
      timeout: 10
      expected_status: 200
      allow_notifications: true
      notify_after: 2
      notify_all_changes: true
      public: true
      redirect: true
    services:
      - name: Network Mesh
        domain: http://linkerd-identity.linkerd.svc.cluster.local:9990/metrics
        <<: *httpservice
      - name: Alerting
        domain: http://prometheus-server.alm.svc.cluster.local:80/-/ready
        <<: *httpservice
      - name: Kubernetes
        domain: https://kubernetes.default.svc.cluster.local:443/readyz
        <<: *httpservice
immutable: false
kind: ConfigMap
metadata:
  creationTimestamp: "2024-01-19T07:33:30Z"
  name: statping-services
  namespace: alm
  resourceVersion: "235778523"
  uid: 03e9cb6d-9086-4f54-8f8c-f2c724227f58
```

Let me know if this helps.

**Comment by florian-besser** [comment]
So it seems I've figured out what caused this; before running `statping-ng` we were running `statping`. This caused some problems with `statping-ng` refusing to start. As this is only a test system I proceeded to wipe the `statping` database, as I was suspecting some incompatible data.

This worked and got `statping-ng` working. I did not, however, clear my browser cache!
After I cleared my cache / disabled caching in the network tools the software was working as intended.

I reckon some "bad files" from `statping` were hanging in the cache and that's what caused issues.
